cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/index.js

```diff
@@ -35,31 +37,49 @@ const browserMap = {
 };
 
 export default class DataPrepBrowser extends Component {
-  constructor(props) {
-    super(props);
-    let store = DataPrepBrowserStore.getState();
-    this.state = {
-      activeBrowser: store.activeBrowser
-    };
-  }
-  componentWillMount() {
-    DataPrepBrowserStore.subscribe(() => {
+  state = {
+    activeBrowser: DataPrepBrowserStore.getState().activeBrowser
+  };
+
+  componentDidMount() {
+    this.storeSubscription = DataPrepBrowserStore.subscribe(() => {
       let {activeBrowser} = DataPrepBrowserStore.getState();
       if (activeBrowser.name && this.state.activeBrowser.name !== activeBrowser.name) {
         this.setState({
           activeBrowser
         });
       }
     });
+    if (typeof this.props.setActiveBrowser === 'function') {
```

Comment:
Here as well. I am not sure why we are calling `setActiveBrowser` again if in the previous `if` condition we check for the active browser from the same store. May be this should be in the else?

Reply:
The previous `if` is in the store subscription, while this one is not, so it will only be called once.